python/triton/language/semantic.py

```diff
@@ -1496,8 +1496,13 @@ def dot(lhs: tl.tensor, rhs: tl.tensor, acc: tl.tensor, input_precision: Optiona
         _0 = builder.get_int32(0)
         ret_scalar_ty = tl.int32
     elif out_dtype.is_bf16():
-        raise ValueError(
-            "out_dtype=bfloat16 is unsupported. Please use out_dtype=float32/float16 and cast with `.to(tl.bfloat16)`")
+        if builder.options.backend_name in ('cuda', 'hip'):
```

Comment:
we don't want to check backends explicitly and in general for portability we shouldn't be doing different things. 
I think you can pattern match in your backend if needed

Reply:
If it is not desirable to store backend specifics in the frontend, then maybe move this exception to a lower level? (After all, this exception also turns out to be a backend specific, right?)

Reply:
that would be one solution, but for the `dot` operation we try to limit it to what is commonly supported on tensorcore hardwares to avoid surprises on performance. bfloat16 accumulator seems like an uncommon feature that would run slow on other HW so I'm not sure we want to expose it in the language right now.

Reply:
Got it. Thanks for the clarification!